api/src/passport.setup.js

```diff
@@ -42,11 +38,23 @@ passport.use(
           name: profile.displayName,
           email: profile.emails[0].value,
           profileImg: profile.photos[0].value,
-        },
-      });
-      //console.log('big access token: ', accessToken);
-      const token = jwt.sign({ accessToken, refreshToken, googleKey: profile.id }, process.env.JWT_SECRET);
-      return done(null, token);
+        }
+      })
+        .then((user) => {
+          if (!user[0].calendarId) {
+            console.log('making calendar for: ', user[0]);
+            let calendar = Calendar.createCalendar(accessToken, refreshToken);
+            Promise.resolve(calendar)
+              .then((calendarObject) => {
+                user[0].calendarId = calendarObject.response.id;
+                return user[0].save();
+              });
+          }
+        })
+        .then(()=> {
+          const token = jwt.sign({ accessToken, refreshToken, googleKey: profile.id }, process.env.JWT_SECRET);
+          return done(null, token);
+        });
     }
   )
```

Comment:
Make sure to add a 'catch' block here, that calls done with the error.

Reply:
I have now added the catch block.